packages/xo-server-sdn-controller/src/index.js

```diff
@@ -622,8 +641,19 @@ class SDNController extends EventEmitter {
     vni,
     encrypted,
     mtu,
+    preferredCenterId,
   }) {
-    const privateNetwork = new PrivateNetwork(this, uuidv4())
+    let preferredCenter
+    if (preferredCenterId !== undefined) {
+      preferredCenter = this._xo.getXapiObject(
+        this._xo.getObject(preferredCenterId, 'host')
+      )
+
+      // Put pool of preferred center first
+      poolIds = [...new Set([preferredCenter.$pool.$id, ...poolIds])]
```

Comment:
If you new that `poolIds` contained `preferredCenter.$pool.$id` and you didn't need to ensure pool ids where uniq, the following code would have been more efficient:

```suggestion
      const i = poolIds.indexOf(preferredCenterId)
      poolIds[i] = poolIds[0]
      poolIds[0] = preferredCenterId
```

Reply:
I need to ensure poolIds are unique unfortunately.

Reply:
Don't you need to do this as well when there is no preferred center?

Reply:
No in this case I don't care about the order of the pools.
The only thing I need is the preferredCenter pool id first.